app/src/main/java/app/foodfinderapp/service/RestaurantService.kt

```diff
@@ -1,5 +1,10 @@
 package app.foodfinderapp.service
 
+import app.foodfinderapp.dto.Restaurant
+
 class RestaurantService {
+    fun getAllRestaurants(): List<Restaurant>? {
+        return getAllRestaurants()
```

Comment:
This is a recursive call, which will result in infinite recursion: a runtime error.